### ISSUE TITLE
Read Timeout with stream_context before 4

### ISSUE THREAD

**Issue by worker24h** [VictoriaMetrics/victorialogs-datasource#117]
![image](https://github.com/user-attachments/assets/26c95265-8316-484b-853c-a53d1e21c9eb)

as above， read timeout (default: 30s) by virctorialogs web console

@hagen1778 @dmitryk-dk @Loori-R I am developing show context of logs pannel  https://github.com/VictoriaMetrics/victorialogs-datasource/issues/41

Cloud you see it ?

**Comment by dmitryk-dk** [comment]
Hi @worker24h ! Need to check your question and why it happens. I suppose this is VMLogs limitation

**Comment by worker24h** [comment]
> Hi @worker24h ! Need to check your question and why it happens. I suppose this is VMLogs limitation

I think syntax is  correct，as follow, it's timeout too.
![image](https://github.com/user-attachments/assets/dc3956bc-11bb-4d0e-88b5-09ba3c2d893a)

**Comment by dmitryk-dk** [comment]
> > Hi @worker24h ! Need to check your question and why it happens. I suppose this is VMLogs limitation
> 
> I think syntax is correct，as follow, it's timeout too. The may be a bug? ![image](https://private-user-images.githubusercontent.com/24838476/388900858-dc3956bc-11bb-4d0e-88b5-09ba3c2d893a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyNzU2NzEsIm5iZiI6MTczMjI3NTM3MSwicGF0aCI6Ii8yNDgzODQ3Ni8zODg5MDA4NTgtZGMzOTU2YmMtMTFiYi00ZDBlLTg4YjUtMDliYTNjMmQ4OTNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDExMzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1MjUxOTk3NGM2Mjg0MjQ4ZDdlZDczMWQxNWNmNzUzMjU2MjNkNmQ5ZjU2MjkzY2JjOTgxYTQyNGE1MzRhZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uxIKt_IKba0Fbig4F-PNC4JZYLI0w8uT3sG1-89FZMg)
> 
> @dmitryk-dk @hagen1778

I think there can be solutions. 

1. Could you check the query in the developer mode `network` tab.? Is there `timoute` query param present?
```
The maximum query execution time is limited by -search.maxQueryDuration command-line flag value. This limit can be overridden to smaller values on a per-query basis by passing the needed timeout via timeout query arg. For example, the following command limits query execution time to 4.2 seconds:
```

Could you increase the `-search.maxQueryDuration` flag to `90s` and check if it helps?

**Comment by worker24h** [comment]
> > > Hi @worker24h ! Need to check your question and why it happens. I suppose this is VMLogs limitation
> 
> > 
> 
> > I think syntax is correct，as follow, it's timeout too. The may be a bug? ![image](https://private-user-images.githubusercontent.com/24838476/388900858-dc3956bc-11bb-4d0e-88b5-09ba3c2d893a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyNzU2NzEsIm5iZiI6MTczMjI3NTM3MSwicGF0aCI6Ii8yNDgzODQ3Ni8zODg5MDA4NTgtZGMzOTU2YmMtMTFiYi00ZDBlLTg4YjUtMDliYTNjMmQ4OTNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDExMzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1MjUxOTk3NGM2Mjg0MjQ4ZDdlZDczMWQxNWNmNzUzMjU2MjNkNmQ5ZjU2MjkzY2JjOTgxYTQyNGE1MzRhZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uxIKt_IKba0Fbig4F-PNC4JZYLI0w8uT3sG1-89FZMg)
> 
> > 
> 
> > @dmitryk-dk @hagen1778
> 
> 
> 
> I think there can be solutions. 
> 
> 
> 
> 1. Could you check the query in the developer mode `network` tab.? Is there `timeout` query param present?
> 
> ```
> 
> The maximum query execution time is limited by -search.maxQueryDuration command-line flag value. This limit can be overridden to smaller values on a per-query basis by passing the needed timeout via timeout query arg. For example, the following command limits query execution time to 4.2 seconds:
> 
> ```
> 
> 
> 
> Could you increase the `-search.maxQueryDuration` flag to `90s` and check if it helps?
> 
> 

I know it cloud be changed for timeout period. 
But I search data in latest 5min and limit 50 , it should response quickly.

**Comment by dmitryk-dk** [comment]
> > > > Hi @worker24h ! Need to check your question and why it happens. I suppose this is VMLogs limitation
> > 
> > 
> > > 
> > 
> > 
> > > I think syntax is correct，as follow, it's timeout too. The may be a bug? ![image](https://private-user-images.githubusercontent.com/24838476/388900858-dc3956bc-11bb-4d0e-88b5-09ba3c2d893a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyNzU2NzEsIm5iZiI6MTczMjI3NTM3MSwicGF0aCI6Ii8yNDgzODQ3Ni8zODg5MDA4NTgtZGMzOTU2YmMtMTFiYi00ZDBlLTg4YjUtMDliYTNjMmQ4OTNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIyVDExMzYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1MjUxOTk3NGM2Mjg0MjQ4ZDdlZDczMWQxNWNmNzUzMjU2MjNkNmQ5ZjU2MjkzY2JjOTgxYTQyNGE1MzRhZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uxIKt_IKba0Fbig4F-PNC4JZYLI0w8uT3sG1-89FZMg)
> > 
> > 
> > > 
> > 
> > 
> > > @dmitryk-dk @hagen1778
> > 
> > 
> > I think there can be solutions.
> > 
> > 1. Could you check the query in the developer mode `network` tab.? Is there `timeout` query param present?
> > 
> > ```
> > 
> > The maximum query execution time is limited by -search.maxQueryDuration command-line flag value. This limit can be overridden to smaller values on a per-query basis by passing the needed timeout via timeout query arg. For example, the following command limits query execution time to 4.2 seconds:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Could you increase the `-search.maxQueryDuration` flag to `90s` and check if it helps?
> 
> I know it cloud be changed for timeout period. But I search data in latest 5min and limit 50 , it should response quickly. Do you think it doesn't timeout , too?

It looks like a bug in the VMLogs. If you can develop the feature while the bug will be fixed it would be cool. 
Also you can create the ticket in the https://github.com/VictoriaMetrics/VictoriaMetrics repository with a description

**Comment by hagen1778** [comment]
Hey @worker24h! This seems suspicious. Are you sure VictoriaLogs is up&running and isn't overwhelmed with other service that is making requests?

**Comment by worker24h** [comment]
> Hey @worker24h! This seems suspicious. Are you sure VictoriaLogs is up&running and isn't overwhelmed with other service that is making requests?

Yes If I remove **stream_context before 4**, then get response quickly. So I think this is a bug with stream_context

**Comment by hagen1778** [comment]
I see. Please, open a bug report then.

**Comment by worker24h** [comment]
> I see. Please, open a bug report then.

I want try to fixed it and interest it, but I doesn't read source code never, so do you offer some articles or debugger method? Thanks

**Comment by worker24h** [comment]
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7637
I create a new issue for this BUG.  

 Is the VictoriaLogs not open source? I can't find the source code for it
@hagen1778 @dmitryk-dk

**Comment by dmitryk-dk** [comment]
> > I see. Please, open a bug report then.
> 
> I want try to fixed it and interest it, but I doesn't read source code never, so do you offer some articles or debugger method? Thanks

Hi! You can check the source code starts from the ProcessQueryRequest 
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vlselect/main.go#L194
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/6c6e469bfb8ad84f4d72412a015f7b7292cd713f/lib/logstorage/pipe_stream_context.go#L669

and you can go up from this function.

**Comment by hagen1778** [comment]
Looks like stream context issue was addressed in vlogs [v1.3.0](https://docs.victoriametrics.com/victorialogs/changelog/#v130). Does it mean this ticket can be closed? Does it unblock us from adding `logs context` feature to the plugin?

**Comment by worker24h** [comment]
Sorry, our project has closed `vmlogs`, because the query context feature does not used for me. Apologies, now I don’t have an environment for testing. But, I will continue to follow it.  Now, I will close the issue.

**Comment by hagen1778** [comment]
Thanks for feedback, and sorry for the negative user experience.
